pcsx2/windows/WinPowerProfile.cpp

```diff
@@ -0,0 +1,46 @@
+/*  PCSX2 - PS2 Emulator for PCs
+*  Copyright (C) 2002-2021  PCSX2 Dev Team
+*
+*  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+*  of the GNU Lesser General Public License as published by the Free Software Found-
+*  ation, either version 3 of the License, or (at your option) any later version.
+*
+*  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+*  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+*  PURPOSE.  See the GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License along with PCSX2.
+*  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "PrecompiledHeader.h"
+#include <powrprof.h>
+
+// Checks the current active power plan
+// If the power plan isn't named 'high performance', put a little message in the console
+// This function fails silently
+void CheckIsUserOnHighPerfPowerPlan()
+{
+	GUID* pPwrGUID;
+	DWORD ret = PowerGetActiveScheme(NULL, &pPwrGUID);
```

Comment:
This leaks memory. As per https://docs.microsoft.com/en-us/windows/win32/api/powersetting/nf-powersetting-powergetactivescheme:
> A pointer that receives a pointer to a GUID structure. Use the LocalFree function to free this memory.

Reply:
Oh, right. I even commented that on my basic sample. Oops.